anyblok/column.py

```diff
@@ -1524,3 +1556,21 @@ def update_table_args(self, registry, Model):
         key.update(str(enum).encode('utf-8'))
         name = self.fieldname + '_' + key.hexdigest() + '_types'
         return [CheckConstraint(constraint, name=name)]
+
+
+@CompareType.add_comparator(String, String)
+@CompareType.add_comparator(String, Selection)
+@CompareType.add_comparator(String, Sequence)
+def compare_strings(col1, col2):
+    if col1.size != col2.size:
+        raise FieldException(
+            "You can't add a foreign key on %r(%d) -> %r(%d)" % (
+                col1.__class__, col1.size, col2.__class__, col2.size))
```

Comment:
raise FieldException(
                "You can't add a foreign key using based String columns with different size "
                "`{model1!s}.{col1!s}` pointing to `{model2!s}.{col2!s}` have "
                " different sizes  {type1!r}({size1:d}) -> ({type2!r}){size2:r}".format(
                    model1=col1.model,
                    col1=col1,
                    model2=col2.model,
                    col2=col2,
                    type1=col1.__class__,
                    type2=col2.__class__
                    size1=col2.size
                    size2=col2.size
                )
            )